### PR TITLE
General-legendary-salary: fix linting 

### PR DESCRIPTION
**Pull request by clottman**
## tl;dr: Things in this PR: 
 - clean up the Popover component
 - new linting configuration that will catch more errors
 - fixes for things the new linting configuration complained about 
 - fix for runtime console warning from Icon component due to recently updated search Icon
 - update eslint & its friends
 - turns on dogfood of server-side linting on Glitch 

## Why did this happen
The linting configuration we were previously using in shared components either excludes some important rules, or was failing silently. I learned this because I shipped a version of shared-components that has a bug that the linter should have caught. 

bug from shared-components

I wasn't excited about playing whack-a-mole to figure out additional useful rules that weren't included in `eslint-config-react-app`, so instead I switched the config to extending `eslint:recommended`, `plugin:jsx-a11y/recommended`,  and `plugin:react/recommended`. This is more in line with what the community site does for linting. (The community site also adds the `airbnb` eslint config, which is more opinionated.)

## Popover bug
~Linting fixes revealed a bug! This is the dream of linters everywhere.~ 

**edited to say:**
I was largely wrong about there being a bug in the Popover component; we were passing one prop to something that was just a styled `div` that didn't take props, which I thought was a bug affecting behavior, but it was not affecting behavior at all. (see edit history of this description for more details on what I was thinking)
**resume non-edited content**

Also, since the Popover story example was using `startOpen=true`, when you load shared-components.glitch.me, the Delete Team element that is the popover example immediately receives focus. In addition to fixing the bug, the example now has `startOpen={false}` so you can load shared-components.glitch.me and see the top of the page.  

## Linting fixes for new rules
- Many display names were added
  - [see slack message for details on why that's awesome](https://fogcreek.slack.com/archives/CJ1G77E87/p1583358032000400) 
  - I did not add a display name, and instead ignored the error temporarily, if the component being complained about was only used in a Story and nowhere else
- a bunch of variables were defined but never used, so the variables got removed

## Server side linting
Got [instructions from Osmose on enabling server-side linting](https://gist.github.com/Osmose/63909d3a9913e176bfef692fa3dc28a7) in the glitch editor. It appears to be working! 
 - this requires using `.eslintrc.json`, not `.eslintrc.js`. 
 - this means we can see errors from our plugins (React & jsx-a11y) in the glitch editor!
 - there will be a slight delay when you start editing your first javascript file before the errors start showing up 
  - other weird things might happen; this is experimental 

Server-side linting, wow!

### CODE REVIEW

**Comment by whimsicallyson** [comment]
> Popover from shared-components is already in use in a few places in community, but we don't use the startOpen prop in any of those instances. 

we _should_ be using it in user-options-pop (ln 166). does this change mean it works now, or is there a change we need to do on the community side to get it functioning properly?

File: lib/popover.js [hunk 1]
```diff
@@ -169,6 +169,14 @@ export const Popover = ({ align, renderLabel, views = {}, initialView, startOpen
     </PopoverContainer>
   );
 };
+
+Popover.defaultProps = {
+  /* there's a good chance this isn't the right default, but it maintains backwards compatability with the community site,
+  from a time when Popover was not actually respecting startOpen and was passing true through to the underlying PopoverContainer
+  all the time */ 
+  startOpen: true,
```
Comment (whimsicallyson) [inline lib/popover.js]: I'm a bit worried about this... since most popovers shouldn't startOpen. If this starts working as intended, will we suddenly have an open popover nightmare site?

Granted that's also something we can probably test locally before updating the package version in prod, right?

File: lib/popover.js [hunk 2]
```diff
@@ -169,6 +169,14 @@ export const Popover = ({ align, renderLabel, views = {}, initialView, startOpen
...
+};
+
 Popover.propTypes = {
   align: PropTypes.oneOf(Object.keys(alignments)).isRequired,
   renderLabel: PropTypes.func.isRequired,
```
Comment (whimsicallyson) [inline lib/popover.js]: add `startOpen: PropTypes.bool`?

File: package.json [hunk 3]
```diff
@@ -35,16 +36,15 @@
...
-    "eslint": "^5.16.0",
-    "eslint-config-react-app": "^4.0.1",
-    "eslint-plugin-flowtype": "^2.50.3",
-    "eslint-plugin-import": "^2.18.0",
-    "eslint-plugin-jsx-a11y": "^6.2.3",
-    "eslint-plugin-react": "^7.14.2",
-    "eslint-plugin-react-hooks": "^1.5.0",
     "express": "4.16.4",
+    "eslint": "^6.8.0",
+    "eslint-config-airbnb": "^18.0.1",
```
Comment (whimsicallyson) [inline package.json]: are we actually using this?

**Comment by clottman** [comment]
Updated to remove my introduction of a bug-fix-that-doesn't-fix-anything-but-maybe-breaks-stuff in the Popover component. 

I thought we were using the community-specific Popover in user-options-pop in Community, but @whimsicallyson pointed out I was wrong on that. We _were_ passing in `startOpen` as always true to PopoverContainer - but that's fine, because PopoverContainer is just a styled div, which doesn't read that prop at all. So, I removed `startOpen` from PopoverContainer inside Popover, but left everything else alone.

Still working on testing these changes- [the instructions in contributing.md](https://github.com/glitchdotcom/shared-components/blob/master/CONTRIBUTING.md#testing-a-version-of-shared-components-in-the-community-stie) aren't working for me :(

**Comment by clottman** [comment]
@whimsicallyson and I both tested this as a pre-release on staging; it works! Deploying now

**Comment by keithk** [comment]
:rocket: PR was released in [`v0.19.0`](https://github.com/glitchdotcom/shared-components/releases/tag/v0.19.0) :rocket:
